File: app/src/main/java/com/yenvth/soilDetectionApp/diction/DictionPresenter.java
package com.yenvth.soilDetectionApp.diction;

import com.yenvth.soilDetectionApp.models.HistoryModel;

public interface DictionPresenter<V extends DictionView> {
    void saveSearchHistory(HistoryModel historyModel);
    void getListSoil(String queryString);
}

File: app/src/main/java/com/yenvth/soilDetectionApp/diction/DictionPresenter.kt
package com.yenvth.soilDetectionApp.diction

import com.yenvth.soilDetectionApp.room.entity.HistoryRecord

interface DictionPresenter<V : DictionView?> {
    fun saveSearchHistory(historyRecord: HistoryRecord)
    fun getListSoil(queryString: String)
}